### PR TITLE
VTL escape javascript 

### PR DESCRIPTION
**Pull request by calvernaz**
This PR tackles the escape javascript function used on VTL templates by not "double" encoding strings.
I also converted a new test to be AWS-validated.
For example,

```
#set( $body = $input.json("$") )
#define( $loop )
{
    #foreach($e in $map.keySet())
       #set( $k = $e )
       #set( $v = $map.get($k))
       // previously $k and $v weren't wrapped on double-quotes, because of the bug, however it's required (validated on AWS)
       "$k": "$v"
       #if( $foreach.hasNext ) , #end
    #end
}
```

or issues like 

```
""proxy"":
""test""
```
It also organises the step functions integration tests using parameters to ensure APIs are deployed between changes. The previous tests was not working on AWS since an update to REST APIs needed to be deployed.

Resolves https://github.com/localstack/localstack/issues/7188

### CODE REVIEW

File: localstack/services/apigateway/templates.py [hunk 1]
```diff
@@ -87,16 +87,15 @@ def urlDecode(self, s):
...
-        return json.dumps(s)
+        """
+        Escapes a string that will turn any regular single quotes (') into escaped ones (\').
+        JSON dumps will escape the single quotes.
+        https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html
+        """
+        # empty string escapes to empty object
+        if isinstance(s, str) and len(s.strip()) == 0:
+            return "{}"
+        return s if is_number(s) else json.dumps(s)[1:-1]
```
Comment (whummer) [inline localstack/services/apigateway/templates.py:98]: I feel like we should consider a few more corner cases here, right? What if `s` is `None`, then the result returned would be the string `"ul"` (trimmed from `"null"`)

Similar for the case where `s` is `True` or `False` - it just happens to be the case that `is_number(...)` also treats booleans as numbers (not really accurate by the way), but would be better to make these cases explicit in here...?

**Review by whummer — COMMENTED** [review]
Nice catch about the incorrect behavior of `escapeJavaScript` @calvernaz ! 👍 Only have a few comments, mainly some questions around handling of edge / corner cases..

File: tests/integration/test_apigateway.py [hunk 2]
```diff
@@ -202,6 +219,7 @@ def test_create_rest_api_with_custom_id(
         assert response.ok
         assert response._content == b'{"echo": "foobar", "response": "mocked"}'
 
+    @pytest.mark.skip
```
[the remark this answers is not in the document]
Comment (calvernaz) [reply tests/integration/test_apigateway.py:222]: Sorry, this shouldn't have been committed. This test has some issues running on Mac (at least to me).

File: tests/unit/test_apigateway.py [hunk 3]
```diff
@@ -208,24 +208,21 @@ def test_render_template_values():
...
-        ('{"foo": 123}', '{"foo": 123}'),
-        ('{"foo"": 123}', '"{\\"foo\\"\\": 123}"'),
-        (1, "1"),
-        (True, "true"),
+        ('""', '\\"\\"'),
+        ('"""', '\\"\\"\\"'),
+        ('{"foo": 123}', '{\\"foo\\": 123}'),
+        ('{"foo"": 123}', '{\\"foo\\"\\": 123}'),
+        (1, 1),
+        (None, None),
```
Comment (whummer) [inline tests/unit/test_apigateway.py:221]: Shouldn't this be
```suggestion
        (None, "null"),
```
?

Comment (calvernaz) [reply tests/unit/test_apigateway.py:221]: The `None` is not a string, so not sure what you are referring to. 

This is what happens on AWS:

Input:
```json
{
    "myAttr": null
}
```

With the transformation template:

```
{
  "result": $util.escapeJavaScript($input.json('$.myAttr'))
}
```

The result:

```
Endpoint request body after transformations: {
  "result": null
}
```

Comment (whummer) [reply tests/unit/test_apigateway.py]: Great, thanks for checking the behavior on real AWS @calvernaz ! 🙌  Again, sorry for not making myself clear - what I wanted to suggest is that we should make `escapeJavaScript` return a `str` in any case (i.e., should not return values like `None`, or the number `0`, etc). 

I've pushed a tiny commit to ensure `escapeJavaScript` returns a string (and also added a corresponding type hint) - hope that's ok for you. Excited to get this merged soon! 🎉

**Review by whummer — APPROVED** [review]
LGTM! 👍 🎉

File: tests/unit/test_apigateway.py (hunk 3 above)
Comment (calvernaz) [reply tests/unit/test_apigateway.py]: I think i did, but the string "null" is not the null javascript value or json fwiw. So IMO that should not be handled by this function but from the caller.

**Pull request merged**